src/js/containers/agency/visualizations/ObligatedContainer.jsx

```diff
@@ -0,0 +1,158 @@
+        super(props);
+
+        this.state = {
+            loading: true,
+            error: false,
+            inFlight: true,
+            obligatedAmount: 0,
+            budgetAuthority: 0,
+            fiscalQuarter: 0,
+            cgacCode: ""
+        };
+
+        this.loadData = this.loadData.bind(this);
+        this.setCgacCode = this.setCgacCode.bind(this);
+        this.setFiscalQuarter = this.setFiscalQuarter.bind(this);
+    }
+
+    componentWillMount() {
+        this.loadData(this.props.id, this.props.activeFY);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.id !== nextProps.id) {
+            this.loadData(nextProps.id, nextProps.activeFY);
+        }
+    }
+
+    setCgacCode(id) {
+        this.cgacRequest = AgencyHelper.fetchAgencyCgacCode({
+            id
+        });
+
+        this.cgacRequest.promise
+            .then((res) => {
+                this.setState(
+                    {
+                        cgacCode: res.data.results[0].toptier_agency.cgac_code
+                    }
+                );
+                this.setFiscalQuarter(this.state.cgacCode);
+            })
+            .catch((err) => {
+                if (!isCancel(err)) {
+                    console.log(err);
+                }
+            });
+    }
+
+    setFiscalQuarter(cgacCode) {
+        this.quarterRequest = AgencyHelper.fetchAgencyFiscalQuarter({
+            cgac_code: cgacCode
+        });
+
+        this.quarterRequest.promise
+            .then((res) => {
+                this.setState({
+                    fiscalQuarter: res.data.results[0].reporting_fiscal_quarter
+                });
+            })
+            .catch((err) => {
+                if (!isCancel(err)) {
+                    console.log(err);
+                }
+            });
+    }
+
+    loadData(agencyID, activeFY) {
+        if (this.searchRequest) {
+            // A request is currently in-flight, cancel it
+            this.searchRequest.cancel();
+        }
+
+        // TODO - Lizzie: uncomment when available
+        this.searchRequest = AgencyHelper.fetchAgencyObligatedAmounts({
+            //fiscal_year: agencyID,
+            //funding_agency_id: activeFY
+            fiscal_year: 2017,
+            funding_agency_id: 246
+        });
+
+        this.setState({
+            inFlight: true
+        });
+
+        this.searchRequest.promise
+            .then((res) => {
+                this.searchRequest = null;
+
+                this.setState({
+                    inFlight: false
+                });
+
+                this.setState({
+                    obligatedAmount: parseFloat(res.data.results[0].obligated_amount),
+                    budgetAuthority: parseFloat(res.data.results[0].budget_authority_amount)
+                });
+
+                // TODO - Lizzie: uncomment when available
+                //this.setCgacCode(this.props.id);
+                this.setCgacCode(246);
```

Comment:
On further investigation, it should be chained together; refactored so chain is explicit rather than nested in function callbacks.